### PR TITLE
New geturl command to download files for LAMMPS inputs on demand

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request adds a new command "geturl" to LAMMPS which can download files via an interface to libcurl.
This could be done with a large variety of protocols supported by libcurl. Applications could be to download data, molecule, or potential files from a repository.

**Related Issue(s)**

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

This is new functionality.

**Implementation Notes**

This implements support for libcurl in a similar fashion to supporting libjpeg or libpng/libz.
Also introspection support and with this library interface support is included

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.

### CODE REVIEW

**Comment by akohlmey** [comment]
@simongravelle Here is how I envision how we could make the tutorial downloads automatic. For each tutorial there would be a base URL and a folder. The folder would contain a manifest file (e.g. a simple manifest.txt file with a list of files). We would download and process the manifest from the GUI and then set up folder(s) and files. I've decided to make this a generic LAMMPS command rather than adding this to the GUI since I see applications for regular LAMMPS use as well.
The KIM package makes even more sophisticated use of libcurl by including database queries to get system properties, but not everybody wants to go through the effort of installing and configuring OpenKIM.

In the GUI we could add a "Tutorials" entry to the main menu and this could contain entries like "Start tutorial 1", "Start tutorial 2" and so on. Those would ask the user to designate a directory for the tutorial and populate it with the initial files and open the initial input in the editor. This doesn't have to be limited to your set of tutorials, so there could be multiple sections and also multiple links to the tutorial online docs.

**Comment by akohlmey** [comment]
@lammps/core any (more) concerns about merging this now?
It would be highly beneficial for me for continuing to adapt LAMMPS-GUI to launch Simon's tutorials ASAP.

File: doc/src/geturl.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Download a file from an URL to the local disk. This is implemented with
+the `libcurl library <https:://curl.se/libcurl/>`_ which supports a
+large variety of protocols including "http", "https", "ftp", "scp",
+"sftp", "file".  The transfer will only be performed on MPI rank 0.
+
+The *output* keyword can be used to set the filename. By default, the last part
+of the URL is used.
+
+The *verify* keyword allows to turn on or off whether ``libcurl`` will validate
```
Comment (sjplimp) [inline doc/src/geturl.rst]: ... keyword sets whether libcurl ...

Comment (sjplimp) [reply doc/src/geturl.rst]: ... keyword determines whether libcurl ...

(more consistent with description of next 2 keywords)

Comment (akohlmey) [reply doc/src/geturl.rst]: done

**Review by sjplimp — APPROVED** [review]

**Pull request merged**